apis/apps/pub/inplace_update.go

```diff
@@ -62,12 +62,21 @@ type InPlaceUpdateState struct {
 	// UpdateEnvFromMetadata indicates there are envs from annotations/labels that should be in-place update.
 	UpdateEnvFromMetadata bool `json:"updateEnvFromMetadata,omitempty"`
 
+	// UpdateResources indicates there are resources that should be in-place update.
+	UpdateResources bool `json:"updateResources,omitempty"`
+
+	// VerticalUpdateOnly indicates there are only vertical update in this revision.
+	VerticalUpdateOnly bool `json:"verticalUpdateOnly"`
+
 	// NextContainerImages is the containers with lower priority that waiting for in-place update images in next batch.
 	NextContainerImages map[string]string `json:"nextContainerImages,omitempty"`
 
 	// NextContainerRefMetadata is the containers with lower priority that waiting for in-place update labels/annotations in next batch.
 	NextContainerRefMetadata map[string]metav1.ObjectMeta `json:"nextContainerRefMetadata,omitempty"`
 
+	// NextContainerResources is the containers with lower priority that waiting for in-place update resources in next batch.
+	NextContainerResources map[string]v1.ResourceRequirements `json:"nextContainerResources,omitempty"`
```

Comment:
Is this to defer the resource updating until the next update operation when both the updated image and the updated resources are present?

Reply:
When resizing resources for a multi-container pod, if we wish to adjust each container individually, it's necessary to track resource changes for containers in subsequent batches.

Reply:
Ok, I got it.